### PR TITLE
[dev-env] e2e Windows tests

### PR DESCRIPTION
**Pull request by pschoffer**
# Description
It would be great to get an automated test that validates dev-env working on windows machines. 

Investigating this I ran into an issue with running Linux based images on `windows-latest` runner. I was not able to figure out how to enable them. So I opened a [github discussion](https://github.com/community/community/discussions/29255)

In the meantime, though I think it would be worth adding the creation test which works fine. The way I coded it is very rough, so I would be happy to get any feedback on it. 

ping @bsessions85

### CODE REVIEW

**Comment by mjangda** [comment]
I looked into this before and one blocker was that GitHub didn't support running docker inside docker. Is that not an issue any more? Or do we have a workaround?

**Comment by pschoffer** [comment]
Hmm I am still fighting with other dev-env issues. But guessing from the fact that windows runner comes with docker and docker-compose preinstalled it might work?

https://github.com/actions/runner-images/blob/main/images/win/Windows2022-Readme.md#tools

@mjangda

**Comment by pschoffer** [comment]
It seems to be able to do docker in docker, but I am having hard time convincing it to run Linux containers as the daemon runs in win architecture. I need to read up and try more stuff in order to figure out how to massage it some more.

**Comment by mjangda** [comment]
This is what I was thinking of:

https://github.com/actions/runner/issues/904

File: __tests__/e2e_test.bat [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+rem dev-env tests
+
+echo "== Creating dev-env"
+
+call vip dev-env create --client-code image --title Test --multisite false --php 8.0 --wordpress 6.0 --mu-plugins image -e false -p false -x false
+
+if NOT %errorlevel% == 0 (
+    echo "== Failed to create dev-env"
+    exit 1
+)
```
Comment (mjangda) [inline __tests__/e2e_test.bat:22]: Do we want to check that the config files for the environment were actually created?

**Review by mjangda — COMMENTED** [review]
One minor note but this is a very nice improvement 👍

File: __tests__/e2e_test.bat (hunk 1 above)
Comment (pschoffer) [reply __tests__/e2e_test.bat:22]: Good idea. Done.

**Review by rinatkhaziev — APPROVED** [review]
Thanks! Admittedly the usefulness is limited because we can't run the containers.
@bsessions85 any ideas on how can we side-step this?

**Comment by bsessions85** [comment]
> Thanks! Admittedly the usefulness is limited because we can't run the containers. @bsessions85 any ideas on how can we side-step this?

Thats a good question @rinatkhaziev. I have made it work with linux environment but haven't tried it on Windows before. I just did some quick searching and it doesn't appear to be possible as things stand today. I'll keep thinking through it, perhaps there is a workaround somehow that I'm not thinking of.

**Pull request merged**